ddtrace/opentracer/tracer.py

```diff
@@ -71,35 +75,123 @@ def __init__(self, service_name=None, config=None, scope_manager=None):
+            one or more parent SpanContexts. (See the Reference documentation
+            for detail)
+        :param tags: an optional dictionary of Span Tags. The caller gives up
+            ownership of that dictionary, because the Tracer may use it as-is
+            to avoid extra data copying.
+        :param start_time: an explicit Span start time as a unix timestamp per
+            time.time()
+        :param ignore_active_span: an explicit flag that ignores the current
+            active `Scope` and creates a root `Span`.
+        :return: an already-started Span instance.
+        """
+        ot_parent = child_of      # 'ot_parent' is more readable than 'child_of'
+        ot_parent_context = None  # the parent span's context
+        dd_parent = None          # the child_of to pass to the ddtracer
+
+        # Okay so here's the deal for ddtracer.start_span:
+        #  - whenever child_of is not None ddspans with parent-child relationships
+        #    will share a ddcontext which maintains a hierarchy of ddspans for
+        #    the execution flow
+        #  - when child_of is a ddspan then the ddtracer uses this ddspan to create
+        #    the child ddspan
+        #  - when child_of is a ddcontext then the ddtracer uses the ddcontext to
+        #    get_current_span() for the parent
+        if ot_parent is None and not ignore_active_span:
+            # attempt to get the parent span from the scope manager
+            scope = self._scope_manager.active
+            parent_span = getattr(scope, 'span', None)
+            ot_parent_context = getattr(parent_span, 'context', None)
+            # we want the ddcontext of the active span in order to maintain the
+            # ddspan hierarchy
+            dd_parent = getattr(ot_parent_context, '_dd_context', None)
+        elif ot_parent is not None and isinstance(ot_parent, Span):
+            # a span is given to use as a parent
+            ot_parent_context = ot_parent.context
+            dd_parent = ot_parent._dd_span
+        elif ot_parent is not None and isinstance(ot_parent, SpanContext):
+            # a span context is given to use to find the parent ddspan
+            dd_parent = ot_parent._dd_context
+        elif ot_parent is None:
+            # user wants to create a new parent span we don't have to do anything
+            pass
+        else:
+            raise TypeError('invalid span configuration given')
+
+        # create a new otspan and ddspan using the ddtracer and associate it with the new otspan
+        otspan = Span(self, ot_parent_context, operation_name)
+        ddspan = self._tracer.start_span(name=operation_name, child_of=dd_parent)
+        ddspan.start = start_time or ddspan.start  # set the start time if one is specified
+        if tags is not None:
+            ddspan.set_tags(tags)
+        otspan._add_dd_span(ddspan)
+
+        # activate this new span
+        self._scope_manager.activate(otspan, False)
```

Comment:
in the `start_span` API OT doesn't activate the current span (but it uses the current active scope): https://github.com/opentracing/basictracer-python/blob/v3.0.0/basictracer/tracer.py#L109-L116

Reply:
👍